Adds a "how to" for routed nic devices on VMs (#12107)
canonical/lxd: doc/reference/devices_nic.md

      169.254.0.1
      fe80::1

>   For VMs, the gateways must be configured manually or via a mechanism like `cloud-init` (see the [how to guide](../howto/instances_routed_nic_vm.md)).

Reviewer: ```suggestion
  For VMs, the gateways must be configured manually or via a mechanism like `cloud-init` (see {ref}`instances-routed-nic-vm`).
```

Author: Fixed.
